src/lib/registration/sw_mgmt.rb

```diff
@@ -123,28 +124,22 @@ def self.find_base_product
       selected = product_selected? if Stage.initial
       installed = product_installed? if Stage.initial
 
+      # list of products defined by the "system-installation()" provides
+      system_products = Y2Packager::ProductReader.installation_package_mapping.keys
```

Comment:
Suggestion: what about moving this logic as a class method to `Y2Packager::Product`? At some extent, `ProductReader` looks like an "implementation detail".

Reply:
Yes, originally I wanted to use `Y2Packager::Product`, but I found some problems with that:

- It does not support all attributes required by registration, e.g. `register_target`
- The registration code expects a `Hash`, the `["name"]` code would have to be changed to `.name` at *very lot* places (or we would need to add a `.to_h` conversion)
- At the end the product is converted to `SUSE::Connect::Remote::Product` before sending to SCC, maybe we could convert and use it earlier ?

That's not trivial so we have to postpone it, I have added at least a `FIXME` note...